component/core/src/test/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImplTest.java

```diff
@@ -507,6 +507,8 @@ public void testGetNewerOnUserSpacesActivities() throws Exception {
       activity.setTitle("activity title " + i);
       activity.setUserId(demoIdentity.getId());
       activityStorage.saveActivity(spaceIdentity, activity);
+      //Wait for 1 ms
```

Comment:
Comments should explain why this line is needed, not what it does (it is obvious that sleep(1) waits for 1ms).

Reply:
I agreed about commenting it with additional information could be helpful for reader, but really, no one knows why it fails randomly. Probably because the database doesn't flushes as fast as in HDD or SSD ( we have remarked the difference between both types of drives )
Anyway, as for thousands of "wait" instructions in ou unit tests, I think that none will be able to give you really the exact answer of "Why wait" nor commenting it in this unit test ;-)

Reply:
ok I see, so I would prefer putting your explication in comments than the current comment. Something like "We don't know exactly why, but this sleep is needed to make the build pass in any case (maybe a database flush fastness issue ?)". Even if it does not explain the real reason, it gives clues for the maintainers who will need to update these test cases in the future, and warn him/her that removing it could cause issues.

Reply:
In fact the build was failing for a very good reason, the spaces/activities creation and updates are very quick that we have 2 activities created in the same millisecond and we have an avatar added then updated in the same millisecond. :o
I added the 1 ms sleep to make created/updated dates different :-)
the tests were written considering the latency of those operations and now they fail.

Reply:
@thomasdelhomenie  I updated the description , is it ok for you ?

Reply:
```suggestion
      //Wait for 1 ms to make sure the activities are not same at the exact same time
```
Here is my suggestion.